nolearn/lasagne/base.py

```diff
@@ -89,7 +89,11 @@ def __iter__(self):
                 yb = self.y[sl]
             else:
                 yb = None
-            yield self.transform(Xb, yb)
+            if self.info is not None:
+                infob = np.array(self.info)[sl]
+            else:
+                infob = None
+            yield self.transform(Xb, yb, infob)
 
     @classmethod
     def _shuffle_arrays(cls, arrays, random):
```

Comment:
I see. I like that.

On Tue, May 24, 2016 at 3:28 PM, Scott Mayer McKinney <
notifications@github.com> wrote:

> In nolearn/lasagne/base.py
> <https://github.com/enlitic/nolearn/pull/4#discussion_r64486301>:
>
> > @@ -89,7 +89,11 @@ def __iter__(self):
> >                  yb = self.y[sl]
> >              else:
> >                  yb = None
> > -            yield self.transform(Xb, yb)
> > +            if self.info is not None:
> > +                infob = np.array(self.info)[sl]
> > +            else:
> > +                infob = None
> > +            yield self.transform(Xb, yb, infob)
> >
> >      @classmethod
> >      def _shuffle_arrays(cls, arrays, random):
>
> for array in arrays:
> if array is None:
> continue
> ...
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/enlitic/nolearn/pull/4/files/f669694b78133861767c141b2166eac17a71a131#r64486301>
>